index.ts

```diff
@@ -0,0 +1,63 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import depd from 'depd';
+
+const deprecate = depd('hsts');
+
+const DEFAULT_MAX_AGE = 180 * 24 * 60 * 60;
+
+interface HstsOptions {
+  includeSubDomains?: boolean;
+  maxAge?: number;
+  preload?: boolean;
+  setIf?: (req: IncomingMessage, res: ServerResponse) => boolean;
+}
+
+function alwaysTrue () {
+  return true;
+}
+
+export = function hsts (options: HstsOptions = {}) {
+  if ('includeSubdomains' in options) {
+    deprecate('The "includeSubdomains" parameter is deprecated. Use "includeSubDomains" (with a capital D) instead.');
+  }
+
+  if ('setIf' in options) {
+    deprecate('The "setIf" parameter is deprecated. Refer to the documentation to see how to set the header conditionally.');
+  }
+
+  const { maxAge = DEFAULT_MAX_AGE, setIf = alwaysTrue } = options;
+
+  if ('maxage' in options) {
```

Comment:
Can we revert this to `options.hasOwnProperty('maxage')`?

Reply:
Sure, though note the verbosity of `Object.prototype.hasOwnProperty.call(..., ...)` due to [this eslint rule](https://eslint.org/docs/rules/no-prototype-builtins)